tests/unit/single_table/test_base.py

```diff
@@ -104,3 +108,584 @@ def test_fit(self):
+        expected_data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.],
+            COND_IDX: [0, 1, 2]
+        })
+        pd.testing.assert_frame_equal(expected_data, result)
+        instance._sample_in_batches.assert_called_once_with(
+            num_rows=3,
+            batch_size=3,
+            max_tries_per_batch=None,
+            conditions=condition,
+            transformed_conditions=transformed_condition,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None
+        )
+
+    def test__conditionally_sample_rows_raises_value_error(self):
+        """Test when sampled rows is lower or 0."""
+        # Setup
+        transformed_data = pd.DataFrame({
+            COND_IDX: [0, 1, 2],
+            'salary.value': [65., 75., 85.]
+        })
+        instance = Mock()
+        instance._sample_in_batches.return_value = []
+        condition = Mock()
+        transformed_condition = 'fancy_condition'
+
+        # Run and Assert
+        expected_message = re.escape(
+            'Unable to sample any rows for the given conditions '
+            '`fancy_condition`. Try increasing `max_tries_per_batch` (currently: None) '
+            'or increasing `batch_size` (currently: 3). Note that '
+            'increasing these values will also increase the sampling time.'
+        )
+        with pytest.raises(ValueError, match=expected_message):
+            BaseSynthesizer._conditionally_sample_rows(
+                instance,
+                transformed_data,
+                condition,
+                transformed_condition,
+                graceful_reject_sampling=False,
+            )
+
+    def test__conditionally_sample_rows_raises_value_error_model_is_gm(self):
+        """Test when sampled rows is lower or 0 and model is ``GaussianMultivariate``."""
+        # Setup
+        transformed_data = pd.DataFrame({
+            COND_IDX: [0, 1, 2],
+            'salary.value': [65., 75., 85.]
+        })
+        instance = Mock()
+        instance._sample_in_batches.return_value = []
+        condition = Mock()
+        transformed_condition = 'fancy_condition'
+        instance._model = GaussianMultivariate()
+
+        # Run and Assert
+        expected_message = re.escape(
+            'Unable to sample any rows for the given conditions '
+            '`fancy_condition`. This may be because the provided values are out-of-bounds in the '
+            'current model. \nPlease try again with a different set of values.'
+        )
+        with pytest.raises(ValueError, match=expected_message):
+            BaseSynthesizer._conditionally_sample_rows(
+                instance,
+                transformed_data,
+                condition,
+                transformed_condition,
+                graceful_reject_sampling=False,
+            )
+
+    def test__conditionally_sample_rows_without_grafecul_reject_sampling(self):
```

Comment:
shouldn't this say `with_graceful_reject_sampling` since it is `True` in this test. also a typo: grafecul